src/main/java/com/carrotcreative/cream/util/HashingUtil.java

```diff
@@ -0,0 +1,52 @@
+package com.carrotcreative.cream.util;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import android.support.annotation.NonNull;
+
+public class HashingUtil
+{
+    private static final int CACHE_SIZE = 50;
+    private static LruCache<String, String> mCache = new LruCache<>(CACHE_SIZE);
+
+    private HashingUtil() {}
+
+    /**
+     *
+     * @param secret The string that needs to be hashed.
+     * @return The hashed value.
+     */
+    public static String getHash(@NonNull String secret)
+    {
+        String hashedValue = mCache.get(secret);
+        if(hashedValue == null)
+        {
+            hashedValue = generateHash(secret);
+            mCache.put(secret, hashedValue);
+        }
+
+        return hashedValue;
+    }
+
+    private static String generateHash(String secret)
+    {
+        String hashedString = null;
+        try
+        {
+            MessageDigest messageDigest = MessageDigest.getInstance("SHA-256");
+            messageDigest.update(secret.getBytes("UTF-8"));
+            hashedString = new String(messageDigest.digest());
```

Comment:
As it turned out, the byte array was not getting converted into a proper string. I tried, ` new String(message.digest(), "UTF-8"); ` but the result was still in raw format and not readable string.

Added a bytesToHex function to convert the bytes to hexadecimal string. Fixed in commit da2e24d257a25b167b30b101383c544f5d18b40e